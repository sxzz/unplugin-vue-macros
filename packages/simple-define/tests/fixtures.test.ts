import { resolve } from 'node:path'
import { describe } from 'vitest'
import {
  RollupEsbuildPlugin,
  RollupRemoveVueFilePathPlugin,
  RollupVue,
  RollupVueJsx,
  rollupBuild,
  testFixtures,
} from '@vue-macros/test-utils'
import VueSimpleDefine from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    ['tests/fixtures/*.vue', '!tests/fixtures/*.exclude.vue'],
    (args, id) =>
      rollupBuild(id, [
        VueSimpleDefine(),
        RollupVue(),
        RollupVueJsx(),
        RollupRemoveVueFilePathPlugin(),
        RollupEsbuildPlugin({
          target: 'esnext',
        }),
      ]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    }
  )
})
