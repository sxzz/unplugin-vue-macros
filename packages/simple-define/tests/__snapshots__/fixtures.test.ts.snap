// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"import { defineComponent, toDisplayString } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"basic\\",
  setup(__props) {
    const props = simpleProps([\\"msg\\"]);
    return (_ctx, _cache) => {
      return toDisplayString(_ctx.msg);
    };
  }
});

var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { basic as default };
"
`;
